peer.go

```diff
@@ -1154,13 +1155,19 @@ out:
 				atomic.StoreInt64(&p.pingLastSend, now)
 			}
 
-			// Write out the message to the socket, closing the
-			// 'sentChan' if it's non-nil, The 'sentChan' allows
+			// Write out the message to the socket, responding with
+			// error if `errChan` is non-nil. The `errChan` allows
 			// callers to optionally synchronize sends with the
 			// writeHandler.
 			err := p.writeMessage(outMsg.msg)
 			if outMsg.errChan != nil {
-				outMsg.errChan <- err
+				// NOTE: A non-blocking select is used in case
+				// this isn't the only error that gets signaled,
+				// and errors are not read from the other end.
+				select {
+				case outMsg.errChan <- err:
```

Comment:
~~I'm not sure this is the best approach, as this could lead to the race where we are eventually going to read from the `errChan`, but haven't gotten around to it at this point. In that case we will lose this (nil-)error.~~

See my other comment, we should be able to know for sure that this is sent on only once.

Reply:
I think the assumption here is that it's possible for a failed broadcast to result in the err chan being sent upon twice.